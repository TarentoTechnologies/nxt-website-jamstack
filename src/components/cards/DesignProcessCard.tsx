import React from "react";

import { carbon } from "../../styles/style-guide/ColorGuide.module.css";
import { h4500 } from "../../styles/style-guide/Typography.module.css";
import {
  cardStyles,
  imgStyles,
  titleStyles,
} from "./DesignProcessCard.module.css";

interface DesignProcessCardProps {
  title: string;
  icon: any;
}

export const DesignProcessCard = ({ title, icon }: DesignProcessCardProps) => {
  return (
    <div
      className={`col-6 col-md-3 ${cardStyles}`}
      data-testid="design-process-card"
    >
      <div className={`${imgStyles}`}>
        <img
          src={icon}
          alt={title}
          title={title}
          className={`img-fluid`}
          data-aos="fade-up"
        />
      </div>
      <h4 className={`${titleStyles} ${h4500} ${carbon}`} data-aos="fade-up">
        {title}
      </h4>
    </div>
  );
};
