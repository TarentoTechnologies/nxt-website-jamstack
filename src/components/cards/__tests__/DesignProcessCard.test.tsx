import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import React from "react";

import { DesignProcessCard } from "../DesignProcessCard";

describe("Test if the DesignProcessCard component is working as expected", () => {
  it("<DesignProcessCard /> matches snapshot", () => {
    const component = render(
      <DesignProcessCard
        title="Discover"
        icon={
          "https://strapi-stage-nxt.tarento.com/uploads/discover_img_4bc56123b0.svg"
        }
      />
    );
    expect(component.container).toMatchSnapshot();
  });

  it("<DesignProcessCard /> renders without exploding", () => {
    const component = render(
      <DesignProcessCard
        title="Discover"
        icon={
          "https://strapi-stage-nxt.tarento.com/uploads/discover_img_4bc56123b0.svg"
        }
      />
    );
    expect(component.getByTestId("design-process-card")).toBeInTheDocument();
  });
});
