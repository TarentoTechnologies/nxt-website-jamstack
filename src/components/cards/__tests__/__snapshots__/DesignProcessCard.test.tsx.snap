// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if the DesignProcessCard component is working as expected <DesignProcessCard /> matches snapshot 1`] = `
<div>
  <div
    class="col-6 col-md-3 cardStyles"
    data-testid="design-process-card"
  >
    <div
      class="imgStyles"
    >
      <img
        alt="Discover"
        class="img-fluid"
        data-aos="fade-up"
        src="https://strapi-stage-nxt.tarento.com/uploads/discover_img_4bc56123b0.svg"
        title="Discover"
      />
    </div>
    <h4
      class="titleStyles h4500 carbon"
      data-aos="fade-up"
    >
      Discover
    </h4>
  </div>
</div>
`;
